lib/web/apiserver_test.go

```diff
@@ -5946,16 +5901,30 @@ func waitForOutput(stream *terminalStream, substr string) error {
 	}
 }
 
-func (s *WebSuite) client(opts ...roundtrip.ClientParam) *client.WebClient {
+func (s *WebSuite) client(t *testing.T, opts ...roundtrip.ClientParam) *TestWebClient {
 	opts = append(opts, roundtrip.HTTPClient(client.NewInsecureWebClient()))
 	wc, err := client.NewWebClient(s.url().String(), opts...)
 	if err != nil {
 		panic(err)
 	}
-	return wc
+	return &TestWebClient{wc, t}
 }
 
-func (s *WebSuite) login(clt *client.WebClient, cookieToken string, reqToken string, reqData interface{}) (*roundtrip.Response, error) {
+type TestWebClient struct {
+	*client.WebClient
+	t *testing.T
+}
+
+func (c *TestWebClient) RoundTrip(fn roundtrip.RoundTripFn) (*roundtrip.Response, error) {
+	c.t.Helper()
+	resp, err := c.WebClient.RoundTrip(fn)
+
+	verifySecurityResponseHeaders(c.t, resp.Headers())
```

Comment:
The potential issue with holding onto `t` like this is that if the client is passed to a subtest, or used on a goroutine other than the one running the test, it will not be correct to call `t.FailNow()` (which is what the `testify/require` functions will do). It may be enough to just have `RoundTrip` return an error if the security headers are not there.

Reply:
Other than that, I think I do like having the check in the client.

Reply:
I was thinking about this and just realized, the other thing to note is that since Go does not do dynamic dispatch, your `(*TestWebClient).RoundTrip` will only run if called directly - if the test just calls `clt.Get` or something else, that will go to the embedded `*client.WebClient` and it will not call back out to anything from `TestWebClient`.

If we think it's worth running this on all client calls, I think you'd either have to wrap each `WebClient` method, or install a custom `http.RoundTripper` to the underlying `http.Client.Transport`

Reply:
Sorry, I should have commented that I was aware of this.  I removed my changes in the service and made sure that this minimal verification was able to verify my changes (ie the tests fail).  I tried several permutations and felt like that even though this was not absolute it was possibly "good enough" while being simple.

If we do want to extend to the other functions I felt like the simplicity was sort of loss and maybe the former implementation was better.  But again, I don't feel strongly one way or another, but it was recognized.

Reply:
Yeah I think it should be good enough, it just broke my original assumption when I realized that so thought I'd comment in case it was missed, glad to know it was considered.

Reply:
I left a comment to make sure there is no confusion and also committed a simplification that was missed earlier.  Thank you for raising it!